core/services/request/files/FileSubmission.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace EventEspresso\core\services\request\files;
+
+use finfo;
+
+/**
+ * Class FileSubmission
+ *
+ * All the info about a file from $_FILES (except mimetype, because that's not reliable), but put together onto one
+ * object with a few helpers.
+ * FilesDataHandler takes care of creating these from $_FILES.
+ *
+ * @package     Event Espresso
+ * @author         Mike Nelson
+ * @since         $VID:$
+ *
+ */
+class FileSubmission implements FileSubmissionInterface
+{
+    /**
+     * @var string original name on the client machine
+     */
+    protected $name;
+
+    /**
+     * @var string mime type
+     */
+    protected $type;
+
+    /**
+     * @var string file extension
+     */
+    protected $extension;
+
+    /**
+     * @var int in bytes
+     */
+    protected $size;
+
+    /**
+     * @var string local filepath to the temporary file
+     */
+    protected $tmp_file;
+
+    /**
+     * @var int one of UPLOAD_ERR_OK, UPLOAD_ERR_NO_FILE, UPLOAD_ERR_INI_SIZE, UPLOAD_ERR_FORM_SIZE or other values
+     * although those aren't expected.
+     */
+    protected $error_code;
+
+    public function __construct($name, $tmp_file, $size, $error_code = null)
+    {
+        $this->name = basename($name);
+        $this->tmp_file = $tmp_file;
```

Comment:
I'm not sure what sanitization could or should be applied to the tmp_file. Suggestions?